src/scalarstats.jl

```diff
@@ -274,7 +274,7 @@ function mad(x; center=nothing, normalize::Union{Bool, Nothing}=nothing, constan
     # Knowing the eltype allows allocating a single array able to hold both original values
     # and differences from the center, instead of two arrays
     S = isconcretetype(T) ? promote_type(T, typeof(middle(zero(T)))) : T
-    x2 = x isa AbstractArray ? LinearAlgebra.copy_oftype(x, S) : collect(S, x)
+    x2 = x isa AbstractArray ? copyto!(similar(x, S), x) : collect(S, x)
```

Comment:
Note that this change assumes that when `S == T`, `copyto!(similar(x, S), x) ` is as fast as `copy(x)`. Not sure whether that holds: I could add another branch to ensure that this simple case is fast.